src/libraries/System.Private.CoreLib/src/System/Text/StringBuilder.cs

```diff
@@ -1179,6 +1179,15 @@ public StringBuilder Append(char value)
                 return this;
             }
 
+            unsafe
+            {
+                Span<char> tempBuffer = stackalloc char[32]; // should be enough for all primitives
```

Comment:
This is insufficient for Guid.

Reply:
Again, nevermind, we don't use this path for Guid; TryFormat is used directly from AppendFormatHelper.